wcivf/apps/people/tests/test_person_views.py

```diff
@@ -179,6 +179,97 @@ def test_one_candidacy_intro(self):
+            election=election,
+            post=post,
+            ballot_paper_id="local.welsh.assembly.2022-05-01",
+        )
+        person_post = PersonPostFactory(
+            person=self.person,
+            election=election,
+            party=party,
+            party_name=party.party_name,
+            post=post,
+            post_election=pe,
+        )
+
+        previous_party_1 = PartyFactory(
+            party_name="Conservative", party_id="foo_id"
+        )
+        previous_party_2 = PartyFactory(party_name="Labour", party_id="bar_id")
+        person_post.previous_party_affiliations.add(previous_party_1)
+        person_post.previous_party_affiliations.add(previous_party_2)
+        self.assertEqual(person_post.previous_party_affiliations.count(), 2)
+        self.assertEqual(self.person.current_or_future_candidacies.count(), 1)
+        response = self.client.get(self.person_url, follow=True)
+
+        self.assertTemplateUsed(
+            "elections/includes/_previous_party_affiliations.html"
+        )
+        self.assertTemplateUsed("people/includes/_person_intro_card.html")
+
+        self.assertContains(
+            response,
+            "has declared their affiliation with the following parties in the past 12 months",
+        )
+
+    def test_previous_party_affiliations_in_past_elections_table(self):
+        """This is a test for current previous party affiliations in the
+        previous elections table"""
+
+        past_election = ElectionFactory(
+            name="Welsh Local Election",
+            current=False,
+            election_date="2021-05-01",
+            slug="local.welsh.assembly.2021-05-01",
+        )
+        party = PartyFactory(
+            party_name="Conservative and Unionist Party", party_id="ConUnion"
+        )
+        past_person_post = PersonPostFactory(
+            person=self.person,
+            post_election__election=past_election,
+            election=past_election,
+            party=party,
+            votes_cast=1000,
+        )
+        previous_party_1 = PartyFactory(
+            party_name="Conservative", party_id="foo_id"
+        )
+        previous_party_2 = PartyFactory(party_name="Labour", party_id="bar_id")
+        past_person_post.previous_party_affiliations.add(previous_party_1)
+        past_person_post.previous_party_affiliations.add(previous_party_2)
+
+        response = self.client.get(self.person_url, follow=True)
+
+        self.assertEqual(self.person.past_not_current_candidacies.count(), 1)
+        self.assertTemplateUsed(
+            "people/includes/_person_previous_elections_card.html"
+        )
+        self.assertTemplateUsed(
+            "elections/includes/_previous_party_affiliations.html"
+        )
+        self.assertContains(
+            response,
+            "Other party affiliations in the past 12 months",
+            html=True,
+        )
+        self.assertContains(response, "Conservative, Labour", html=True)
+
```

Comment:
One additional test that would be useful would be to check that when the person doesn't have any previous party affiliations, that the related text is **not** shown